swp/migrations/0040_publication_publication_dt.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 3.2.16 on 2023-04-13 14:56
+
+from django.db import migrations, models
+
+from swp.utils.date import parse_date
+
+
+def migrate_date(apps, schema_editor):
+    Publication = apps.get_model('swp', 'Publication')
+
+    for publication in Publication.objects.all():
+        default = publication.created.date().isoformat()
+        publication.publication_dt = parse_date(publication.publication_date, default_date_str=default)
```

Comment:
Following my initial discussion about losing the information whether a publication date string could be (fully parsed): your field is nullable, maybe keep it null, when no date could be parsed instead of filling it with a misleading date derived from the created date.